components/console-backend-service/internal/domain/eventing/resolver.go

```diff
@@ -0,0 +1,202 @@
+package eventing
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/kyma-project/kyma/components/console-backend-service/internal/gqlschema"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+)
+
+type TriggerList []*v1alpha1.Trigger
+
+func (l *TriggerList) Append() interface{} {
+	e := &v1alpha1.Trigger{}
+	*l = append(*l, e)
+	return e
+}
+
+func (r *Resolver) TriggersQuery(ctx context.Context, namespace string, subscriber *duckv1.Destination) ([]*v1alpha1.Trigger, error) {
+	items := TriggerList{}
+	var err error
+
+	if subscriber == nil {
+		err = r.Service().ListInNamespace(namespace, &items)
+	} else if subscriber.Ref != nil {
```

Comment:
which index should be used when Ref and URI is set? maybe it is not a case, I don't know the trigger domain and we can skip it

Reply:
Me neither, but we could always introduce a combined index for both ref and URI, just like in hostname+service index in apigateway.